wgpu-core/Cargo.toml

```diff
@@ -83,41 +94,58 @@ glsl = ["naga/glsl-in"]
 ## Enable `ShaderModuleSource::SpirV`
 spirv = ["naga/spv-in", "dep:bytemuck"]
 
+#! ### Other
+# --------------------------------------------------------------------
+
+## Internally count resources and events for debugging purposes. If the counters
+## feature is disabled, the counting infrastructure is removed from the build and
+## the exposed counters always return 0.
+counters = ["wgpu-types/counters"]
+
 ## Implement `Send` and `Sync` on Wasm, but only if atomics are not enabled.
-##
-## WebGL/WebGPU objects can not be shared between threads.
-## However, it can be useful to artificially mark them as `Send` and `Sync`
-## anyways to make it easier to write cross-platform code.
-## This is technically *very* unsafe in a multithreaded environment,
-## but on a wasm binary compiled without atomics we know we are definitely
-## not in a multithreaded environment.
 fragile-send-sync-non-atomic-wasm = [
     "wgpu-hal/fragile-send-sync-non-atomic-wasm",
-    "wgpu-types/fragile-send-sync-non-atomic-wasm",
 ]
 
-#! ### Backends, passed through to wgpu-hal
+#! ### External libraries
 # --------------------------------------------------------------------
+#! The following features facilitate integration with third-party supporting libraries.
 
-## Enable the `metal` backend.
-metal = ["wgpu-hal/metal"]
+## Enable using the `mach-dxcompiler-rs` crate to compile DX12 shaders.
+static-dxc = ["wgpu-hal/static-dxc"]
 
-## Enable the `vulkan` backend.
-vulkan = ["wgpu-hal/vulkan"]
-
-## Enable the `GLES` backend.
-##
-## This is used for all of GLES, OpenGL, and WebGL.
-gles = ["wgpu-hal/gles"]
-
-## Enable the `dx12` backend.
-dx12 = ["wgpu-hal/dx12"]
+#! ### Target Conditional Features
+# --------------------------------------------------------------------
+# Look to wgpu-hal's Cargo.toml for explaination how these features and the wgpu-core
+# platform crates collude to provide platform-specific behavior.
+
+## DX12 backend
+dx12 = ["wgpu-core-platform-windows-linux-android-emscripten/dx12"]
+## Metal backend
+metal = ["wgpu-core-platform-apple/metal"]
+## Vulkan backend, only available on Windows, Linux, Android
+vulkan = ["wgpu-core-platform-windows-linux-android-emscripten/vulkan"]
+## OpenGL backend, only available on Windows, Linux, Android, and Emscripten
+gles = ["wgpu-core-platform-windows-linux-android-emscripten/gles"]
+
+## WebGL backend, only available on Emscripten
+webgl = ["wgpu-core-platform-wasm/webgl"]
+## OpenGL backend, on macOS only
+angle = ["wgpu-core-platform-apple/angle"]
+## Vulkan portability backend, only available on macOS
+vulkan-portability = ["wgpu-core-platform-apple/vulkan-portability"]
+## Renderdoc integration, only available on Windows, Linux, and Android
+renderdoc = ["wgpu-core-platform-windows-linux-android-emscripten/renderdoc"]
 
 [dependencies]
 naga.workspace = true
 wgpu-hal.workspace = true
```

Comment:
gosh this is the real genius of this solution: on my attempt I had to juggle with forwardings from the proxy crate, but at this level we already **know** that we'll need wgpu-hal, we just don't know which features to activate yet (which is then handled by the proxy crates)

I think we should note this down somewhere so future generations won't try to do it at the `wgpu` level

Reply:
Added some docs on wgpu-hal's Toml